### DISCUSSION TITLE
Help regarding Meaning of Usage of Scaling while defining the displacement variables in MOOSE SolidMechanics Physics type problem

### DISCUSSION THREAD

**Discussion by shrituntunroy** [idaholab/moose#28453] [Q&A Modules: Solid mechanics] [answered]
Hi Moose Team,

I am a new Moose user and currently learning to use Moose.

I have a doubt regarding defination of variables in the Moose input file.

Ref : https://github.com/apatra6/rhocp/blob/main/examples/tantalum/strain_rate_effect/1e-4ps/bcc_pxtal.i

```
[Variables]
  [./disp_x]
    order = FIRST
    family = LAGRANGE
    scaling = 1e-4
  [../]
  [./disp_y]
    order = FIRST
    family = LAGRANGE
    scaling = 1e-4
  [../]
  [./disp_z]
    order = FIRST
    family = LAGRANGE
    scaling = 1e-4
  [../]
[]

```

I am unable to understand what is the function of scaling parameter while defining the variables for FE problem.

I have referred the MOOSE documentation and source files still didn't understood the purpose of scaling parameter in the defination of variables.

And How do we decide ideal value of scaling parameter.

Kindly help me understanding this concept.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Scaling is used as a simple preconditioning of the equations. When there are multiple equations, we weight the residuals of the equation for each variable by this scaling factor to make the convergence criterion more fair and/or help with matrix conditioning

This page on porous flow has an explanation
https://mooseframework.inl.gov/modules/porous_flow/convergence.html#46b5c8c6-401d-44a8-80b8-1ae589dcd695

> And How do we decide ideal value of scaling parameter.

Experimentally most of the time. You look at the residuals and see if they are orders of magnitude apart, maybe it's worth bringiing them closer together.

**Reply by shrituntunroy** [reply]
Hi Guillaume,

Thank you for your response.

From what I understand, the scaling parameter is particularly useful when solving multiphysics problems in MOOSE. Would it also be beneficial when dealing with single-physics problems, such as mechanics, in MOOSE?

Thank you.

**Reply by GiudGiud** [reply]
It would because the displacements (the equation for each component) might have very different residuals still. For example a traction test might have much larger residual in the direction of the traction

**Comment by shrituntunroy** [comment]
Thankyou  Guillaume, for your help
